packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1196,6 +1196,8 @@ class TextSelectionGestureDetectorBuilder {
   void onDragSelectionUpdate(DragStartDetails startDetails, DragUpdateDetails updateDetails) {
     if (!delegate.selectionEnabled)
       return;
+    if (!editableText.widget.focusNode.hasFocus)
```

Comment:
I would prefer remove these lines:
https://github.com/flutter/flutter/blob/ca14d85cf7f983ce038f0f385283e314ea55afdf/packages/flutter/lib/src/widgets/editable_text.dart#L2344-L2346

 instead of explicitly preventing drag update events from getting through when the text field is unfocused. Even if the text field is unfocused you may still want edge scroll?

Unfortunately some tests that use `EditableText` are using tap to focus the input field and removing the said lines breaks those tests but that should be easy to fix? Not sure how many tests are doing that though.

Reply:
Oh another thing is current the text selection highlight is only visible if the text field has the focus. Maybe we should change that so when the selection is being changed the highlight is still visible.

Reply:
@vs-krasnov Do you have any opinion on this as the original author of the issue?  What if the field unfocused but the selection continued to be visible and to change with the drag?

Reply:
@justinmc thanks for taking care of this issue so quickly ❤️ 
IMHO, once a text field becomes unfocused, we shouldn't change the state of this text field anymore based on future drag events.
I wouldn't expect the selection to continue for unfocused text fields.

Reply:
It looks like Chrome does continue to allow edge scrolling but doesn't show the highlight:

![web](https://user-images.githubusercontent.com/389558/144500296-f3cd36cf-39a4-47a2-9360-33e44e1e41e0.gif)

Does it work for everyone if we go with that behavior in Flutter?  I've updated the PR to do that.  Assuming tests go green.

<details>

<summary>HTML example code</summary>

```html
<!doctype html>
<html class="no-js" lang="">

<head>
  <meta charset="utf-8">
  <title></title>
  <meta name="description" content="">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <meta property="og:title" content="">
  <meta property="og:type" content="">
  <meta property="og:url" content="">
  <meta property="og:image" content="">

  <link rel="manifest" href="site.webmanifest">
  <link rel="apple-touch-icon" href="icon.png">
  <!-- Place favicon.ico in the root directory -->

  <link rel="stylesheet" href="css/normalize.css">
  <link rel="stylesheet" href="css/main.css">

  <meta name="theme-color" content="#fafafa">

  <style>
    .box {
      width:  100px;
      height: 100px;
      background-color: dodgerblue;
    }
  </style>
</head>

<body>

  <!-- Add your site or application content here -->
  <p>Hello world! This is HTML5 Boilerplate.</p>
  <input type="number" />
  <div class="box">Hover over this to unfocus</div>
  <script src="js/vendor/modernizr-3.11.2.min.js"></script>
  <script src="js/plugins.js"></script>
  <script src="js/main.js"></script>

  <!-- Google Analytics: change UA-XXXXX-Y to be your site's ID. -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-XXXXX-Y', 'auto'); ga('set', 'anonymizeIp', true); ga('set', 'transport', 'beacon'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async></script>
  <script>
    document.querySelector('.box').addEventListener('mouseenter', () => {
      document.activeElement.blur();
    });
  </script>
</body>

</html>
```

</details>

Reply:
It makes sense to follow the existing platform's behavior 👍